### PR TITLE
[MM-51878]: Migrate billing history unit test to testing library

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Migrate billing history unit test to testing library

#### Ticket Link
 Fixes https://mattermost.atlassian.net/browse/MM-51878
  
#### Screenshots
NA

#### Release Note
```release-note

-  Migrate billing history unit test to testing library

```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/billing/billing_history.test.tsx [hunk 1]
```diff
@@ -99,12 +97,20 @@ describe('components/admin_console/billing/billing_history', () => {
...
+        renderWithIntl(
             <Provider store={store}>
                 <BillingHistory/>
             </Provider>,
         );
-        expect(wrapper).toMatchSnapshot();
+
+        expect(screen.queryByText('Billing History')).toBeInTheDocument();
+        expect(screen.queryByText('Transactions')).toBeInTheDocument();
+        expect(screen.queryByText('All of your invoices will be shown here')).toBeInTheDocument();
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/billing/billing_history.test.tsx:108]: What you are doing with these migrations is fantastic and will help us in going off enzyme later on

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/admin_console/billing/billing_history.test.tsx (hunk 1 above)
Comment (neallred) [reply webapp/channels/src/components/admin_console/billing/billing_history.test.tsx:108]: ^ this

**Review by neallred — APPROVED** [review]
LGTM, thanks @AshishDhama !

File: webapp/channels/src/components/admin_console/billing/billing_history.test.tsx (hunk 1 above)
Comment (saturninoabril) [reply webapp/channels/src/components/admin_console/billing/billing_history.test.tsx:108]: I totally agree!

File: webapp/channels/src/components/admin_console/billing/billing_history.test.tsx [hunk 2]
```diff
@@ -99,12 +97,20 @@ describe('components/admin_console/billing/billing_history', () => {
     const store = mockStore(state);
 
     test('should match snapshot', () => {
```
Comment (saturninoabril) [inline webapp/channels/src/components/admin_console/billing/billing_history.test.tsx:99]: Maybe the test title should be changed as well as it no longer do snapshot?

File: webapp/channels/src/components/admin_console/billing/billing_history.tsx [hunk 3]
```diff
@@ -39,6 +39,7 @@ export const NoBillingHistorySection = (props: NoBillingHistorySectionProps) =>
             />
         </div>
         <ExternalLink
+            data-testid='BillingHistory-link'
```
Comment (saturninoabril) [inline webapp/channels/src/components/admin_console/billing/billing_history.tsx:42]: I know we're not consistent but our convention for `data-testid` and element ID are `camelCase`. Would be great if this and others here are updated.

**Review by saturninoabril — APPROVED** [review]
Thanks @AshishDhama! Overall LGTM, except for few requests but are non-blocking.

**Review by saturninoabril — APPROVED** [review]
Thanks @AshishDhama!

**Pull request merged**